src/components/common/forms/SignUpForm.tsx

```diff
@@ -0,0 +1,123 @@
+import React from "react";
+import { useForm } from "react-hook-form";
+import { ErrorMessage } from "@hookform/error-message";
+import { useNavigate } from "react-router-dom";
+
+import { useAppSelector, useAppDispatch } from "@/state/app/hooks";
+import { setUsername } from "@/state/features/user";
+import { Button, Input, Dropdown } from "@components/common";
+import { signUpRequest } from "@/services";
+import { urlFormat } from "@/utils";
+import { Pages } from "@/pages";
+
+import styles from "./Forms.module.css";
+
+const SignUpForm = () => {
+  const { genders } = useAppSelector((state) => state.user);
+  const dispatch = useAppDispatch();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitted, isValid },
+    setError,
+    clearErrors,
+  } = useForm();
+
+  const navigate = useNavigate();
+  const isFormValid = !isSubmitted || isValid;
+
+  const handleErrorMessage = (() => {
+    if (errors?.password?.type === "minLength") {
+      return (
+        <ErrorMessage
+          errors={errors}
+          name="password"
+          message="Password should be at least 6 characters long"
+          render={({ message }) => <>{message}</>}
+        />
+      );
+    } else if (!errors?.serverError) {
+      return "All fields are required";
+    }
+  })();
+
+  const onSubmit = async (data: object) => {
+    try {
+      const response = await signUpRequest({
+        user: data,
+      });
+      const { user } = response;
+      dispatch(setUsername(user.name));
+      navigate(urlFormat(Pages.EmailConfirmation));
+      setTimeout(() => {
+        navigate(urlFormat(Pages.ConfirmationDone));
+      }, 1500);
```

Comment:
Maybe it's good to add a comment here like: "TODO: add confirmation email instead of timeout"

Reply:
Perfect, added!